### DISCUSSION TITLE
Hotswap speaker weights during inference with same base model?

### DISCUSSION THREAD

**Discussion by michaellin99999** [espnet/espnet#5718] [Q&A]
Hi,
I have finetuned two versions of ESPNet myself.  one with a mans voice and one with a womans, the base model is the same based on VITS.

Is there a way where I can hotswap weights during runtime?  
the experience I am looking for is I only need to host 1 model. and can simulate a man, woman talking with each other jsut by hotswapping the weights during inference?  preferably less than 2 second latency. I am currently running on a 4090 GPU.

**Comment by sw005320** [comment]
How about loading two models and preparing two instances of `Text2Speech` with these different models instead of swapping the weights?

**Reply by michaellin99999** [reply]
I'm looking into hot swap during inference considering scale. hosting one model for every user is resource intensive.

**Reply by sw005320** [reply]
OK.
We don't prepare such a function, and I recommend implementing it yourself.
